coremltools/converters/mil/debugging_utils.py

```diff
@@ -170,6 +177,6 @@ def replace_inputs(func, input_vars):
         PASS_REGISTRY["common::dead_code_elimination"](prog)
 
     prog.skip_all_passes = True
-    submodel = ct.convert(prog, convert_to=backend, compute_units=model.compute_unit)
+    submodel = ct.convert(prog, convert_to=backend, compute_units=model.compute_unit, minimum_deployment_target=func.opset_version)
```

Comment:
Not directly related but fixes an error I get with my example conversion script:

>`RuntimeWarning: You will not be able to run predict() on this Core ML model. Underlying exception message was: Error compiling model: "compiler error: Error reading protobuf spec. validator error: Description of multiarray feature 'x_cast_fp16' has FLOAT16 dataType, which is only valid in specification version >= 7. This model has version 6".`

Happy to put this up as a separate PR if that's preferred.

Reply:
This is a correct fix :)